packages/vm/src/backend.rs

```diff
@@ -127,6 +127,13 @@ pub trait Storage {
 pub trait Api: Copy + Clone + Send {
     fn canonical_address(&self, human: &HumanAddr) -> BackendResult<CanonicalAddr>;
     fn human_address(&self, canonical: &CanonicalAddr) -> BackendResult<HumanAddr>;
+
+    fn secp256k1_verify(
+        &self,
+        message_hash: &[u8],
+        signature: &[u8],
+        public_key: &[u8],
+    ) -> BackendResult<()>;
 }
```

Comment:
This trait has a bit of a naming flaw since it has the same name as the one in cosmwasm-std, but is something completely different. Here we have an API exposed by the chain to the VM.

This trait should not contain `secp256k1_verify`. Instead the `Api` from cosmwasm-std should get the method, which is then implemented by calling the VM.

Reply:
Yeah. I'll fix it.

Reply:
Better to rename this (in another PR) to `BlockchainApi` or so.

Short / generic names like `Api` and `Backend` can be very convenient from an usability perspective, but can be confusing. Internally (i.e. not contract devs facing) it would be better IMO to use somewhat longer, more descriptive names.

Reply:
Yes, I agree. Let's reserve the short names for contract developers and use a bit more verbosity here. `BlockchainApi` is nice.

Reply:
https://github.com/CosmWasm/cosmwasm/issues/774